forem/forem
# Fix storybook theme switcher

## What type of PR is this? (check all applicable)


## Description

The theme switcher in our storybook was working locally, but in production wasn't having any effect. Digging into it a bit and this was due to the path `themes/${themeName}` not being available in production. Changing the path to `/stylesheets/themes..` also didn't help because the asset file is a `scss` file and can't be referenced directly in a `link` tag in the HTML `<head>`.

I noticed we had some unused storybook scripts relating to creating these static CSS theme sheets, but they weren't actually being called. In the end, calling this script was all that was needed to generate the CSS themes. I've also renamed the script to be clearer about what it's used for.

I've removed some of the other script names that referenced the same code - I couldn't find any use of them in the codebase so I think this is safe.

## Related Tickets & Documents

N/A

## QA Instructions, Screenshots, Recordings

To check this is working locally:
- Run `yarn storybook` locally
- Go to a component page (e.g. Button)
- Play with the theme switcher dropdown and check the colours update in the UI

To check this will work in production:
- Run `yarn build-storybook`
- Run `npx http-server ./app/javascript/storybook-static`
- Follow the link in the terminal output to view the served up production build at port 8080
- Go to a component page (e.g. Button)
- Play with the theme switcher dropdown and check the colours update in the UI

### UI accessibility concerns?

N/A

## Added/updated tests?

- [X] No, and this is why: documentation change

## [Forem core team only] How will this change be communicated?

_Will this PR introduce a change that impacts Forem members or creators, the
development process, or any of our internal teams? If so, please note how you
will share this change with the people who need to know about it._

      [Admin Guide](https://admin.forem.com/), or
      [Storybook](https://storybook.forem.com/) (for Crayons components)
      [`CHANGELOG.md`](https://github.com/forem/forem/tree/main/CHANGELOG.md)
      or in a [forem.dev](http://forem.dev) post
      replace this line with details on why this change doesn't need to be
      shared_
